cranelift-codegen/src/isa/x86/enc_tables.rs

```diff
@@ -123,6 +123,21 @@ fn size_plus_maybe_sib_or_offset_for_inreg_1(
     sizing.base_size + if needs_sib_or_offset { 1 } else { 0 }
 }
 
+/// Calculates the size while inferring if the first input register (inreg0) requires a dynamic REX
+/// prefix and if the second input register (inreg1) requires a SIB or offset.
+fn size_plus_maybe_sib_or_offset_inreg1_plus_rex_prefix_for_inreg0(
+    sizing: &RecipeSizing,
+    enc: Encoding,
+    inst: Inst,
+    divert: &RegDiversions,
+    func: &Function,
+) -> u8 {
+    let needs_rex = (EncodingBits::from(enc.bits()).rex_w() != 0)
+        || test_input(0, inst, divert, func, is_extended_reg);
```

Comment:
I think this also needs a `|| test_input(1, inst, divert, func, is_extended_reg)` in the case that the base register is `r8-r15`? I have admit I'm not sure which order registers are specified but the other part of the [recipe this is used in](https://github.com/bytecodealliance/cranelift/blob/ae77d49fa49c7dccceb16f0a71380a25cf45531a/cranelift-codegen/meta/src/isa/x86/recipes.rs#L1506) has me suspect. I'm also not sure how well our tests cover getting less-likely encodings like that - I'd not think about it if I didn't just try to do `[r11]` a few days ago.